### ISSUE TITLE
Provide build process for standalone bundle

### ISSUE THREAD

**Issue by jvandemo** [jvandemo/generator-angular2-library#2]
A library can easily be imported using `import` statements.

But how can we bundle a library as a standalone file that we can e.g. include from a CDN using a `<script>` element?

E.g. Angular publishes:
https://code.angularjs.org/2.0.0-alpha.46/angular2.js

When looking at the source code, the last few lines in the file automatically register the library:

``` javascript
System.register("angular2/angular2", ["angular2/common", "angular2/core", "angular2/profile", "angular2/lifecycle_hooks", "angular2/bootstrap", "angular2/upgrade"], true, function(require, exports, module) {
  var global = System.global,
      __define = global.define;
  global.define = undefined;
  function __export(m) {
    for (var p in m)
      if (!exports.hasOwnProperty(p))
        exports[p] = m[p];
  }
  __export(require("angular2/common"));
  __export(require("angular2/core"));
  __export(require("angular2/profile"));
  __export(require("angular2/lifecycle_hooks"));
  __export(require("angular2/bootstrap"));
  __export(require("angular2/upgrade"));
  global.define = __define;
  return module.exports;
});
```

What is the best way to publish such a distributable library file?

**Comment by cthrax** [comment]
Should a library provide such a file?

In my experience, when a library provides such a file, it has to either a) bundle all dependencies in such a file b) have no dependencies or c) provide instructions for including dependencies in the page as well. 

It seems to me this is one of the exact problems that AMD, SystemJS, webpack and associated technologies are solving. Enabling a consumer to create a single file, or multiple files, as fits their needs without re-introducing old problems. Only new problems. :)

**Comment by jvandemo** [comment]
@cthrax &mdash; So then you would always load a library via a package manager and loader (such as JSPM and SystemJS)?

Do you think there is a way or need to also support including a library via a simple `<script>` element?

Thank you for sharing your thoughts :+1:

**Comment by cthrax** [comment]
@jvandemo for my use cases, absolutely. I've even gone so far as to create build scripts that create a "poor man's" package manager to create vendor bundles, with everything concatenated in the correct order, when libraries aren't provided in a package manager consumable format.

It is probably not necessary, but just to elaborate on the type of scenarios I'm referring to with dependencies. Let's say Library A depends on lodash 3.x, Library B depends on lodash 4.x and Library C depends on underscore. If I'm using a package manager, I just override all of those dependencies to the underscore compatible version of lodash 4.x. Now everything works. If Libraries A and B bundle lodash, then I have a version conflict and a very difficult time loading the A and B at the same time. The two ways I've seen to work around this is to 1) require the consumer to pull in the library, ensuring it's in a script tag before library A and B or 2) Pull just the pieces of lodash needed to make the library work and bake it into calls within the library, thereby losing the benefit of maintenance on the lodash project. 

This problem obviously compounds when there is more than one dependency overlapping and that's really what package managers are solving.

The point you raise about CDN loading gave me pause, so I investigated the jspm workflow for dealing with CDN hosted modules. As near as I can tell, that is something supported out of the box with jspm and webpack. Given this and given the bundling capabilities of package managers, I can't think of a use case for a sling <script> library include within an angular2 app. It's quite likely I'm missing something though, so if you have one I'd love to hear it.

**Comment by jvandemo** [comment]
@cthrax &mdash; Interesting thoughts. I have never tried installing a CDN hosted library using jspm. Would that require a "self-executing" bundle that registers the library using `System.register`?

Thanks for your input!

**Comment by cthrax** [comment]
@jvandemo ack, sorry I missed you reply! Actually, it's part of the 0.17 features http://jspm.io/0.17-beta-guide/custom-package-resolutions.html . You basically set a custom resolution and jspm ignores that and the browser config sets up where to look for that resolution. In theory, it doesn't even have to be a CDN, but that seems most likely.

**Comment by jvandemo** [comment]
Thank you, it seems that Angular has made a switch to Webpack for bundling. More investigation is needed. If anyone else has additional input, feel free to add it right here.

**Comment by caroso1222** [comment]
In my opinion, bundling a library for Angular would completely miss the point of, for example, tree shaking in Webpack. This feature hugely improves the compiled bundle by removing non used deps through dependency graph analysis. You would end up loading a bunch of deps you might not need if loading via `<script>`. Say, for example, d3. You can include the whole library via `<script>`, or you can import only `d3-interpolate`.

**Comment by jvandemo** [comment]
@caroso1222 &mdash; True that, we are waiting for official recommendations from the Angular team to hear their opinion on the subject as well. Thank you for providing your input 👍

**Comment by jvandemo** [comment]
I have just released v8.1.0, which generates a Flat ES Module, in alignment with the latest Angular Package Format.

See #60 for more information.
